src/Compiler/Checking/AttributeChecking.fs

```diff
@@ -244,104 +247,154 @@ let private CheckCompilerFeatureRequiredAttribute (g: TcGlobals) cattrs msg m =
     | Some([ILAttribElem.String (Some featureName) ], _) when featureName = "RequiredMembers" ->
         CompleteD
     | _ ->
-        ErrorD (ObsoleteError(msg, m))
+        ErrorD (ObsoleteDiagnostic(true, "", msg, "", m))
 
-/// Check IL attributes for 'ObsoleteAttribute', returning errors and warnings as data
-let private CheckILAttributes (g: TcGlobals) isByrefLikeTyconRef cattrs m =
+let private CheckILObsoleteAttributes (g: TcGlobals) isByrefLikeTyconRef cattrs m =
+    let extractILAttribValueFrom name namedArgs   =
+        match namedArgs with 
+        | ExtractILAttributeNamedArg name (AttribElemStringArg v) -> v 
+        | _ -> ""
     let (AttribInfo(tref,_)) = g.attrib_SystemObsolete
     match TryDecodeILAttribute tref cattrs with
-    | Some ([ILAttribElem.String (Some msg) ], _) when not isByrefLikeTyconRef ->
-            WarnD(ObsoleteWarning(msg, m))
-    | Some ([ILAttribElem.String (Some msg); ILAttribElem.Bool isError ], _) when not isByrefLikeTyconRef ->
+    | Some ([ILAttribElem.String (Some msg) ], namedArgs) when not isByrefLikeTyconRef ->
```

Comment:
Is the following correct?
There are 3 constructors in total https://learn.microsoft.com/en-us/dotnet/api/system.obsoleteattribute?view=net-9.0 , deciding on IsError and Message properties.

None of them should be searched if `isByrefLikeTyconRef` holds.

DiagnosticId and UrlFormat properties can only be set using setters (named args).


If yes => Can the code path for handling isByrefLikeTyconRef,DiagnosticId and UrlFormat be unified here?

Reply:
Should be better now.